src/sqlite3gen.cpp

```diff
@@ -520,72 +901,118 @@ static int step(SqlStmt &s,bool getRowId=FALSE, bool select=FALSE)
   return rowid;
 }
 
-static int insertFile(const char* name)
+static int insertPath(QCString name, bool local=TRUE, bool found=TRUE, int type=1)
 {
   int rowid=-1;
   if (name==0) return rowid;
 
-  bindTextParameter(files_select,":name",name);
-  rowid=step(files_select,TRUE,TRUE);
+  name = stripFromPath(name);
+
+  bindTextParameter(path_select,":name",name.data(),FALSE);
+  rowid=step(path_select,TRUE,TRUE);
   if (rowid==0)
   {
-    bindTextParameter(files_insert,":name",name);
-    rowid=step(files_insert,TRUE);
+    bindTextParameter(path_insert,":name",name.data(),FALSE);
+    bindIntParameter(path_insert,":type",type);
+    bindIntParameter(path_insert,":local",local?1:0);
+    bindIntParameter(path_insert,":found",found?1:0);
+    rowid=step(path_insert,TRUE);
   }
   return rowid;
 }
 
-static int insertRefid(const char *refid)
+static void recordMetadata()
 {
-  int rowid=-1;
-  if (refid==0) return rowid;
+  bindTextParameter(meta_insert,":doxygen_version",versionString);
+  bindTextParameter(meta_insert,":schema_version","0.2.0"); //TODO: this should be a constant somewhere; not sure where
```

Comment:
Not sure where you'd prefer a constant with this schema version (or if you'd prefer I leave it as a literal for now).